File: python/004.py
def palindrome(n):
	forward = n
	reverse = 0
	while n > 0:
		digit = n % 10
		n /= 10
		reverse = reverse * 10 + digit
	return forward == reverse

largest = 0
for a in range(100, 1000):
	for b in range(100, 1000):
		product = a * b
		if product > largest and palindrome(product):
			largest = product
print(largest)

File: java/004.java
class Problem004 {
	private static boolean palindrome(int n) {
		int forward = n, reverse = 0;
		while (n > 0) {
			reverse = 10 * reverse + n % 10;
			n /= 10;
		}
		return forward == reverse;
	}

	public static void main(String[] args) {
		int largestPalindrome = 0;
		for (int a = 100; a < 1000; a++) {
			for (int b = a; b < 1000; b++) {
				int product = a * b;
				if (product > largestPalindrome && palindrome(product)) {
					largestPalindrome = product;
				}
			}
		}
		System.out.println(largestPalindrome);
	}
}
